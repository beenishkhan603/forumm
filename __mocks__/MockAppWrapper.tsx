import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
