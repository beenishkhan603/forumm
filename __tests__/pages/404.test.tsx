import { render } from '@testing-library/react'
import NotFoundPage from 'pages/404'
import MockAppWrapper from '__mocks__/MockAppWrapper'

it('renders not found page unchanged', () => {
  const { container } = render(
    <MockAppWrapper>
      <NotFoundPage />
    </MockAppWrapper>
  )
  expect(container).toMatchSnapshot()
})
