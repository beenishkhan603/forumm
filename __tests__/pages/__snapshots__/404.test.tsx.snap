// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not found page unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex flex-col items-center"
    style="color: rgb(24, 26, 32);"
  >
    <div
      class="text-5xl text-white text-center mt-64 max-w-2xl"
      style="color: rgb(24, 26, 32);"
    >
      Sorry, we couldn't find what you were looking for! ðŸ˜¢
    </div>
    <div
      class="text-xl text-white text-center mt-4 mb-16"
      style="color: rgb(24, 26, 32);"
    >
      404 Not Found
    </div>
    <a
      class="
        bg-forumm-blue-2 text-white relative         
        btn hover:animate-heartbeat rounded-full flex justify-center items-center text-center px-6 py-4 text-base h-8 my-0 undefined"
      href="/"
    >
      Back to main page
    </a>
  </div>
</div>
`;
